app/src/main/kotlin/com/quittle/a11yally/view/ButtonSwitch.kt

```diff
@@ -175,4 +157,26 @@ class ButtonSwitch : LinearLayoutCompat {
             }
         }
     }
+
+    private fun setBackground() {
```

Comment:
While it's not the worst idea to do this refactor, was there some warning that led you to do this? Was it the function being overly long that it warned about refactoring stuff out?

Reply:
Yes, the function was too long and detekt highlighted this and one more occurrence in AccessibilityAnalyzer as ComplexMethod issues.